addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import requests
+
+from odoo.addons.pos_paytm import utils as paytm_utils
+from odoo.exceptions import UserError
+from odoo import fields, models, api, _
+from datetime import datetime
+from dateutil import tz
+
+_logger = logging.getLogger(__name__)
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    paytm_mid = fields.Char(string='PayTM Merchant ID')
+    paytm_tid = fields.Char(string='PayTM Terminal ID')
+    merchant_key = fields.Char(string='PayTM Merchant Key')
```

Comment:
You have to be very careful not to put in your model keys that all the terminals will need, or type of payment, or test mode, etc...
Because if you change one of these values you will have to do it for each of your payment methods. If you have 1 or 2 terminals ok but beyond that it will be very problematic, not efficient, source of error and problem.
You should create a payment_paytm module (a payment provider) to fill in this information in unique ways and if necessary modify it only there (look at the Stripe module)
In addition the payTM provider offers online payments, so your payment provider could be used for website with the same credentials and use webhook at this point.
The idea is to have all the information in a unique way in one place

Reply:
@qle-odoo Currently online payment is blocked by RBI (Reserve bank of India) so we can't use it.
only payment from machine is supported.
And one machine is only link with one POS so what you suggest in this case?

Reply:
Currently, online payment is blocked by RBI (Reserve Bank of India), so we cannot use it.
only payment from the machine is supported.
And a machine is only tied to one outlet, so what would you suggest in that case?

Online payment ok, but one day it will be possible I imagine. It is easier to predict this now than to have to change your model in the future.

A machine is linked to a pos ok.
But several machines for a pos not?
2 cashier on the same posconfig with 2 terminals.

In addition, payment methods are created at the company level, not by point of sale.

So I advise you to create a small payment_paytm module with your few credentials and use them for your payment methods.
the idea is to have the least information to put for your payment methods